### PR TITLE
Specify when DbDataReader.GetSchemaTable returns null

### PR DESCRIPTION
**Pull request by roji**
For context see https://github.com/dotnet/runtime/issues/509, https://github.com/dotnet/runtime/pull/41082

### CODE REVIEW

File: xml/System.Data.Common/DbDataReader.xml [hunk 1]
```diff
@@ -1894,12 +1894,13 @@ private static void GetCredits(String connectionString)
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>Returns a <see cref="T:System.Data.DataTable" /> that describes the column metadata of the <see cref="T:System.Data.Common.DbDataReader" />.</summary>
+        <summary>
+            Returns a <see cref="T:System.Data.DataTable" /> that describes the column metadata of the <see cref="T:System.Data.Common.DbDataReader" />.</summary>
+
+            Returns <see langword="null" /> if the executed command returned no resultset, or after <see cref="M:System.Data.DbDataReader.NextResult" /> returns <see langword="false" />.
```
Comment (carlossanlop) [inline xml/System.Data.Common/DbDataReader.xml:1900]: Here and elsewhere: The new sentence is not inside xml elements:

```suggestion
        <summary>Returns a <see cref="T:System.Data.DataTable" /> that describes the column metadata of the <see cref="T:System.Data.Common.DbDataReader" />.
Returns <see langword="null" /> if the executed command returned no resultset, or after <see cref="M:System.Data.DbDataReader.NextResult" /> returns <see langword="false" />.</summary>
```

File: xml/System.Data/IDataReader.xml [hunk 2]
```diff
@@ -189,7 +189,10 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>Returns a <see cref="T:System.Data.DataTable" /> that describes the column metadata of the <see cref="T:System.Data.IDataReader" />.</summary>
+        <summary>
+            Returns a <see cref="T:System.Data.DataTable" /> that describes the column metadata of the <see cref="T:System.Data.IDataReader" />.</summary>
+
+            Returns <see langword="null" /> if the executed command returned no resultset, or after <see cref="M:System.Data.IDataReader.NextResult" /> returns <see langword="false" />.
```
Comment (carlossanlop) [inline xml/System.Data/IDataReader.xml:195]:
```suggestion
        <summary>Returns a <see cref="T:System.Data.DataTable" /> that describes the column metadata of the <see cref="T:System.Data.IDataReader" />.

Returns <see langword="null" /> if the executed command returned no resultset, or after <see cref="M:System.Data.IDataReader.NextResult" /> returns <see langword="false" />.</summary>
```

**Comment by carlossanlop** [comment]
@TianqiZhang @joelmartinez  do you know why the build passed for this change, considering the new text was outside xml elements?

**Comment by joelmartinez** [comment]
@carlossanlop The pipeline (for the most part) really only processes the elements that it is familiar with ... so in case like this where it encounters free-floating text, it just skips over it since it doesn't know what to do with it.

**Comment by carlossanlop** [comment]
@roji can you please take a look again at this PR? I left some suggestions. There are also branch conflicts to resolve.

**Comment by roji** [comment]
@carlossanlop sorry for leaving this PR for so long. I've merged the suggested fix and rebased on the latest master, resolving conflicts. There's a mergeability warning above which seems odd, though.

**Review by David-Engel — APPROVED** [review]

**Review by cheenamalhotra — APPROVED** [review]

**Comment by roji** [comment]
@carlossanlop I think this is now ready for merging (am not sure if your process requires one of you to do it etc.).

**Pull request merged**